src/Analyzers/CSharp/CodeFixes/UseUTF8StringLiteral/UseUTF8StringLiteralCodeFixProvider.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseUTF8StringLiteral
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseUTF8StringLiteral), Shared]
+    internal class UseUTF8StringLiteralCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        private const char QuoteCharacter = '"';
+        private const string Suffix = "u8";
+
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public UseUTF8StringLiteralCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.UseUTF8StringLiteralDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(CodeAction.Create(
+                CSharpAnalyzersResources.Use_UTF8_string_literal,
+                c => FixAsync(context.Document, context.Diagnostics[0], c),
+                nameof(CSharpAnalyzersResources.Use_UTF8_string_literal)),
+                context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            foreach (var diagnostic in diagnostics)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var arrayNode = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
```

Comment:
Worth adding tests for this scenario, I think. (Stealing from compiler tests 😄 )

```csharp
using System.Collections;
using System.Collections.Generic;

class C : IEnumerable<int>
{
    void M(C c)
    {
        // Each literal of the three is a separate IArrayCreationOperation
        // Lowered code is similar to:
        /*
            C c = new C();
            c.Add(new byte[] { 65 });
            c.Add(new byte[] { 66 });
            c.Add(new byte[] { 67 });
        */
        c = new() { 65, 66, 67 };
        
                    c = new C();
            c.Add(new byte[] { 65 });
            c.Add(new byte[] { 66 });
            c.Add(new byte[] { 67 });
    }

    public void Add(params byte[] bytes)
    {
    }

    public IEnumerator<int> GetEnumerator()
    {
        throw new System.NotImplementedException();
    }

    IEnumerator IEnumerable.GetEnumerator()
    {
        throw new System.NotImplementedException();
    }
}
```

Reply:
Stop it, you're scaring me.

I haven't used IOperation before and now I never want to use it again!

Reply:
> Stop it, you're scaring me.
> 
> I haven't used IOperation before and now I never want to use it again!

It's a great tool, but.. yeah sometimes there are annoying and unexpected edge cases.

I think most of the time the edge cases are those with `IsImplicit`. So, looking those up with Regex similar to `IArrayCreationOperation.+?IsImplicit` (thanks to the way the tests are written) in [grep.app](https://grep.app/search?q=IArrayCreationOperation.%2B%3FIsImplicit&regexp=true&filter[path][0]=src/Compilers/CSharp/Test/IOperation/IOperation/) is good to know what those cases look like.

We could skip analysis for implicit operations to let all those edge cases go away. Tests may still be good for coverage and to demonstrate what cases we don't care about.

Reply:
> So, looking those up with Regex similar to `IArrayCreationOperation.+?IsImplicit` in grep.app

Amazing!!

I regret the fact that I will soon forget this pro-tip.

Reply:
This actually works fine with the code as is, and becomes `c = new() { "A"u8", "B"u8, "C"u8 }` :D

The regular param array scenario is trickier though, as the syntax node for the operation is the whole invocation, but I think its still good to support it, so `M(65, 66, 67)` becomes `M("ABC"u8)`.

I think